scripts/04-effect-for-beginners.md

```diff
@@ -0,0 +1,1622 @@
+
+We'll start with the basics and then move on to slowly build an example program piece by piece, which by the end will feature concurrent asynchronous work, granular error handling, dependency injection and benchmarking.
+
+If you are still curious about why Effect is worth learning, feel free check out my other videos on Effect linked in the cards now, and come back after.
+
+---
+
+Sponsored By: \
+The Effect Team
+
+notes:
+Effect is an amazing library and I want to quickly thank the Effect team for financially supporting the creation of content around it. Now on to the tutorial!
+
+---
+```
+npm install effect
+yarn add effect
+pnpm add effect
+```
+```ts
+import { Effect } from "effect"
+```
+notes:
+First, make sure to add the effect package to your project using your package manager of choice
+with that out of the way lets get to business
+
+---
+
+```ts
+type Effect<Requirements, Error, Value> = (
+  requirements: Requirements
+) => Value| Error;
+```
+
+notes:
+We'll begin with the Effect type. This type is the center of the enitre Effect ecosystem.
+
+The Effect type is generic over 3 parameters which describe its behavior. Requirements, Error, and Value.
+
+You should conceptualize an Effect as a function that takes in its requirements as an argument and returns either a value or an error.
+
+Its important to emphasize this function-like nature of an Effect. Just like a function, an Effect is only a description of its behavior when run. The output value is yielded only when the Effect is ran.
+
+---
+
+The unit `Effect`
+```ts
+type Unit = Effect<never, never, void>;
+```
+
+notes:
+The unit effect is the most basic Effect type. It describes a program that has requirements of type never, meaning it does not have any dependencies. It has a error also of type never, meaning it will never error, and it has a value of type void, meaning it does not return anything.
+
+---
+
+A simple `Effect`
+```ts
+type Simple = Effect<never, Error, number>;
+```
+
+notes:
+Lets break down this effect. Like the unit effect, its requirements field is never, meaning it does not have any dependencies. Unlike the unit effect, its Error type is Error, this means that this Effect may fail with an Error object. The final value type is number, meaning this Effect will yield a number if it succeeds.
+
+---
+
+Creating Effects
+```ts
+// type: Effect<never, never, number>
+const success = Effect.succeed(42);
+
+// type: Effect<never, Error, never>
+const failure = Effect.fail(new Error())
+```
+
+notes:
+To create an Effect from a value we can use the succeed and fail constructor functions. Succeed takes a value and returns an Effect that can only ever succeed with that value. Along the same lines, fail takes a value and returns an Effect that can only ever fail with that value.
+
+---
+Effects compose together
+```ts
+function divide(x: number, y: number): Effect.Effect<never, Error, number> {
```

Comment:
Aside: here the body is eagerly evaluated and also if the type isn't specify we find ourselves with a union, while this is ok we may want to use `Effect.suspend(() => y === 0 ? fail : succeed)` so that the check happens when the Effect is executed and not when the function is invoked and the type is unified automatically

Reply:
I am aware of this, my thinking was that this is the 3rd minute of the video. If your brand new to effect, although you could probably infer, you really don't understand what `suspend` means/does. Whereas the if statement, while not really being 'idiomatic' effect code, makes the point clear and understandable, they can learn about the other functions later.

I'll go ahead and use whatever example you think is best though just let me know.